dipy/viz/actor.py

```diff
@@ -64,46 +67,39 @@ def slicer(data, affine=None, value_range=None, opacity=1.,
     else:
         nb_components = 1
 
-    if value_range is None:
-        vol = np.interp(data, xp=[data.min(), data.max()], fp=[0, 255])
-    else:
-        vol = np.interp(data, xp=[value_range[0], value_range[1]], fp=[0, 255])
-    vol = vol.astype('uint8')
-
+    # Allocate VTK Image Data
     im = vtk.vtkImageData()
-    if major_version <= 5:
-        im.SetScalarTypeToUnsignedChar()
-    I, J, K = vol.shape[:3]
+    if nb_components == 1:
+        if value_range is None:
+            value_range = (np.min(data), np.max(data))
+        else:
+            # Rescale data
+            data[data < value_range[0]] = value_range[0]
```

Comment:
`data` is going to be modified in-place. This is an unexpected behaviour. If we are going to clamp the data, we should do it on the copy hold by the `vtkImageActor`.